examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -0,0 +1,431 @@
+        """save numpy arrays to path/to/results/{name}.npz"""
+        parameter_update_path = path_to_results.joinpath(name)
+        print(f'saving parameter updates to {parameter_update_path}')
+        np.savez(
+            file=parameter_update_path,
+            before=self.before,
+            after=self.after,
+            gradient=self.gradient,
+            update=self.update,
+        )
+
+
+def _update_in_place(pred, grads, label,
+                     handle_types_to_update=[nonbonded.AM1CCCHandler, nonbonded.LennardJonesHandler]):
+    """
+    Notes
+    -----
+    * Currently hard-codes l1 loss -- may want to later break this up
+
+    TODO: check if it's too expensive to do out-of-place updates with a copy
+    """
+
+    # TODO: refactor loss definition out of this function
+    loss = np.abs(pred - label)
+    unit = "kJ/mol"
+
+    message = f"""
+    pred:   {pred:.3f} {unit}
+    target: {label:.3f} {unit} 
+    loss:   {loss:.3f} {unit}
+    """
+    print(message)
+    # TODO: save these also to log file
+
+    dl_dpred = np.sign(pred - label)
+
+    # compute updates
+    parameter_updates = dict()
+
+    for handle_type in handle_types_to_update:
+        # TODO: move this out of update function to help with saving and logging, and eventually move the chain-rule
+        #  step out of manually written function
+        dl_dp = dl_dpred * grads[handle_type]  # chain rule
+        # TODO: save dl_dp, update, and params to disk
+
+        update = _clipped_update(dl_dp, step_sizes[handle_type], gradient_clip_thresholds[handle_type])
+        print(f'incrementing the {handle_type.__name__} parameters by {update}')
+
+        # TODO: define a dict mapping from handle_type to forcefield.q_handle.params or something...
+        if handle_type == nonbonded.AM1CCCHandler:
+            before = np.array(forcefield.q_handle.params)  # make a copy
+            forcefield.q_handle.params += update
+            after = np.array(forcefield.q_handle.params)  # make a copy
+        elif handle_type == nonbonded.LennardJonesHandler:
+            before = np.array(forcefield.lj_handle.params)  # make a copy
+            forcefield.lj_handle.params += update
+            after = np.array(forcefield.lj_handle.params)  # make a copy
+        else:
+            raise (RuntimeError("Attempting to update an unsupported ff handle type"))
+
+        parameter_updates[handle_type] = ParameterUpdate(before, after, dl_dp, update)
+
+        # not sure if I want to print these...
+        # print("before: ", before)
+        # print("after: ", after)
+
+    # TODO: also save dl_dp for the other parameter types we're not necessarily updating, for later analysis
+    #   (would be easier with syntax like forcefield[handle_type])
+
+    return parameter_updates
+
+
+def _save_forcefield(filename):
+    # TODO: update path
+
+    with open(path_to_results.joinpath(filename), 'w') as fh:
+        fh.write(step_params)
+
```

Comment:
Note that this is referencing a global variable (defined in the ```__name__ ==  "__main__"```) block, it may be slightly safer/easier to read to just pass this in directly as an argument to the ```_save_forcefield``` function.

Reply:
Woops, good catch! This was an oversight -- addressed in https://github.com/proteneer/timemachine/pull/331/commits/f0de0da68ae7b63706f481f9af00e7bbdf5c6567